src/python/pants/binaries/binary_tool.py

```diff
@@ -5,23 +5,26 @@
 from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
                         unicode_literals, with_statement)
 
+import os
+
+from pants.base.exceptions import TaskError
 from pants.binaries.binary_util import BinaryUtil
 from pants.subsystem.subsystem import Subsystem
-from pants.util.memo import memoized_method
+from pants.util.memo import memoized_method, memoized_property
 
 
 class BinaryToolBase(Subsystem):
   """Base class for subsytems that configure binary tools.
 
-  Typically, a specific subclass is created via create_binary_tool_subsystem_cls() below.
-  That subclass can be further subclassed, manually, e.g., to add any extra options.
+  Subclasses can be further subclassed, manually, e.g., to add any extra options.
 
   :API: public
   """
   # Subclasses must set these to appropriate values for the tool they define.
   # They must also set options_scope appropriately.
   support_dir = None
```

Comment:
I think the whole point was to get rid of this field, no? Since it is now always `'bin/{}'.format(self._get_name())`

Reply:
Oh, I misinterpreted your change to BuildozerBinary -- I think the right thing for now is probably to do exactly that, but allow subclasses to specify a complete `support_dir` as with (current) buildozer if necessary.

Reply:
How about a get_support_dir() classmethod that returns  'bin/{}'.format(cls._get_name()) by default but can be overridden in specific subclasses?

Reply:
Somehow forgot to refresh and missed your comment again -- I did this in `_resolve_binaryutil_supportdir()` in c60d531.